### DISCUSSION TITLE
Highlight executable files for filesystem source

### DISCUSSION THREAD

**Discussion by miki725** [nvim-neo-tree/neo-tree.nvim#1183] [Q&A]
Is it possible to highlight executable files in the tree view somehow? Not seeing anything in the default options https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/defaults.lua to support that but maybe I missed something?

**Comment by cseickel** [comment]
There's no built-in feature for this, no. If you just wanted to affect the highlight groups used by the filename, you can replace the `name` component with your own version that uses a different highlight group for executable files.

Here is an example of how to write a wrapper around the builtin `name` component so that you can customize the output.
https://github.com/nvim-neo-tree/neo-tree.nvim/discussions/681#discussioncomment-5429393

In your case, you want to replace `result.highlight` instead of `result.text`
